### PR TITLE
feat: update to spring 3

### PR DESCRIPTION
**Pull request by aludwiko**
References #1374

### CODE REVIEW

**Comment by octonato** [comment]
Hmm, we need to rebase this PR with `main` and update the kotlin archetype as well.

File: maven-java/kalix-spring-boot-archetype/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -282,7 +282,7 @@
     <dependency>
       <groupId>org.springframework.boot</groupId>
       <artifactId>spring-boot-starter-test</artifactId>
-      <version>2.7.0</version>
+      <version>3.0.1</version>
```
Comment (efgpinto) [inline maven-java/kalix-spring-boot-archetype/src/main/resources/archetype-resources/pom.xml:285]: I suppose there's a reason for not using 3.0.2 here as well?

**Review by efgpinto — APPROVED** [review]
LGTM with a question.

File: maven-java/kalix-spring-boot-archetype/src/main/resources/archetype-resources/pom.xml (hunk 1 above)
Comment (octonato) [reply maven-java/kalix-spring-boot-archetype/src/main/resources/archetype-resources/pom.xml:285]: Should be 3.0.2

Comment (octonato) [reply maven-java/kalix-spring-boot-archetype/src/main/resources/archetype-resources/pom.xml:285]: Actually, I had already noticed that 3.0.2 was out and was planning to update it. I will do it now.

**Pull request merged**